tests/publiccloud/cloud_netconfig.pm

```diff
@@ -12,77 +12,119 @@
+        my $eth0_public_ip = $instance->ssh_script_output('curl --fail --interface eth0 -s https://wtfismyip.com/text');
+        my $eth1_public_ip = $instance->ssh_script_output('curl --fail --interface eth1 -s https://wtfismyip.com/text');
+        assert_not_equals($eth0_public_ip, $eth1_public_ip, "The connection from eth0 shouldn't have the same public IPv4 address as connection from eth1");
+
+        record_info('Rules', 'Make sure there are IP rules for each IP address');
+        die('There is no IP rule') if ($instance->ssh_script_output("ip rule list all from $local_eth0_ip | wc -l") == 0);
+        die('There is no IP rule') if ($instance->ssh_script_output("ip rule list all from $local_eth0_secondary_ip | wc -l") == 0);
+        die('There is no IP rule') if ($instance->ssh_script_output("ip rule list all from $local_eth1_ip | wc -l") == 0);
+        die('There is no IP rule') if ($instance->ssh_script_output("ip rule list all from $local_eth1_secondary_ip | wc -l") == 0);
+
+        record_info('Neighbors', 'Make sure there is at least one ARP neighbor on each interface');
+        die('There are no ARP neighbors on eth0') if ($instance->ssh_script_output("ip neighbor show dev eth0 | wc -l") == 0);
+        die('There are no ARP neighbors on eth1') if ($instance->ssh_script_output("ip neighbor show dev eth1 | wc -l") == 0);
+
+        record_info('Local removal', 'Remove eth0 secondary address and eth1 default route and check if it reappears.');
+        $instance->ssh_assert_script_run("sudo ip addr del $local_eth0_secondary_ip/24 dev eth0");
+        $instance->ssh_assert_script_run("sudo ip route del default dev eth1");
+        record_info('debug', $instance->ssh_script_output('ip addr show; ip route show'));
+
+        sleep(90);
+        $instance->ssh_assert_script_run("ip addr show dev eth0 secondary");
+        die('There is no secondary address on eth0') if ($instance->ssh_script_output(qq(ip -4 -o a s dev eth0 secondary | grep -Po "inet \\K[\\d.]+" | wc -l)) == 0);
+        die('There is no default route on eth1') if ($instance->ssh_script_output("ip route show default dev eth1 | wc -l") == 0);
+
+        record_info('Provider removal', 'Remove eth0 secondary address and eth1 default route and check if it reappears.');
+        my $resource_group = $provider->get_resource_group_from_terraform_show();
+        record_info('debug', script_output("az network nic list --resource-group $resource_group"));
+        my $nic_name = script_output(qq(az network nic list --resource-group $resource_group | jq -r '.[]|select(.primary==false).name'));
+        my $ipConfig_name = script_output(qq(az network nic list --resource-group $resource_group | jq -r '.[]|select(.primary==false).ipConfigurations[]|select(.privateIPAddress=="$local_eth1_secondary_ip").name'));
+        assert_script_run("az network nic ip-config delete -g $resource_group -n $ipConfig_name --nic-name $nic_name");
+        sleep(90);
```

Comment:
Please add a comment explaining why the sleep is needed.

Reply:
I replaced it with `systemctl start cloud-netconfig.service` so please recheck